helix-term/src/ui/picker.rs

```diff
@@ -266,16 +311,12 @@ impl<T: Item + 'static> Component for FilePicker<T> {
     }
 
     fn cursor(&self, area: Rect, ctx: &Editor) -> (Option<Position>, CursorKind) {
-        self.picker.cursor(area, ctx)
+        let (picker_area, _) = layout_picker(&area, self.picker.show_preview, ctx);
+        self.picker.cursor(picker_area, ctx)
     }
 
     fn required_size(&mut self, (width, height): (u16, u16)) -> Option<(u16, u16)> {
-        let picker_width = if width > MIN_AREA_WIDTH_FOR_PREVIEW {
-            width / 2
-        } else {
-            width
-        };
-        self.picker.required_size((picker_width, height))?;
+        self.picker.required_size((width, height))?;
```

Comment:
I am not sure: It works for me as is, but I have not understood `required_size` nor `cursor` above and why that is even needed.

How can I calculate the right value here? I have no access to the configuration that tells me whether to lay out vertically or horizontally:-/

Reply:
This could be wrong, indeed, but it works for me. But I admit that I am rather fuzzy about what this method is used for.

How would I get access to the Configuration to pick the layout direction out of in this place?